de-lib/src/main/java/org/iplantc/de/fileViewers/client/callbacks/EnsemblUtil.java

```diff
@@ -33,60 +29,49 @@
+             GWT.<EnsemblUtilAppearance>create(EnsemblUtilAppearance.class));
     }
 
-    EnsemblUtil(final File file,
-                final String infoType,
+    EnsemblUtil(final List<DiskResource> resourcesToSend,
                 final IsMaskable container,
+                final DiskResourceUtil diskResourceUtil,
                 final EnsemblUtilAppearance appearance) {
-        this.file = file;
+        this.resourcesToSend = resourcesToSend;
         this.container = container;
-        this.infoType = infoType;
         this.appearance = appearance;
-        this.diskResourceUtil = DiskResourceUtil.getInstance();
+        this.diskResourceUtil = diskResourceUtil;
     }
 
     public void sendToEnsembl(final DiskResourceServiceFacade diskResourceServiceFacade) {
-        List<HasPath> list = new ArrayList<>();
-        final HasPaths diskResourcePaths =
-                diskResourceServiceFacade.getDiskResourceFactory().pathsList().as();
-        final String path = file.getPath();
-        String filename = diskResourceUtil.parseNameFromPath(path);
-        String parent = diskResourceUtil.parseParent(path);
-        String indexFile = null;
-        String indexFilePath = null;
-        if (infoType.equals(InfoType.BAM.toString())) {
-            indexFile = filename + ".bai";
-        } else if (infoType.equals(InfoType.VCF.toString())) {
-            indexFile = filename + ".tbi";
-        } else if (infoType.equals(InfoType.GFF.toString()) || infoType.equals(InfoType.BED.toString())
-                   || infoType.equals(InfoType.BIGBED.toString())
-                   || infoType.equals(InfoType.BIGWIG.toString())) {
-            indexFile = null;
-        }
+        final FastMap<TYPE> pathMap = new FastMap<>();
 
-        list.add(file);
+        for (DiskResource resource : resourcesToSend) {
+            final String path = resource.getPath();
+            final String infoType = resource.getInfoType();
 
-        if (indexFile != null) {
-            indexFilePath = parent + "/" + indexFile;
-            list.add(CommonModelUtils.getInstance().createHasPathFromString((indexFilePath)));
+            pathMap.put(path, TYPE.FILE);
 
-        }
+            String indexFilePath = null;
+            if (InfoType.BAM.toString().equals(infoType)) {
+                indexFilePath = path + ".bai";
+            } else if (InfoType.VCF.toString().equals(infoType)) {
+                indexFilePath = path + ".tbi";
+            }
 
-        if (!Strings.isNullOrEmpty(indexFilePath)) {
-            diskResourcePaths.setPaths(Arrays.asList(path, indexFilePath));
-        } else {
-            diskResourcePaths.setPaths(Arrays.asList(path));
+            if (!Strings.isNullOrEmpty(indexFilePath)) {
+                pathMap.put(indexFilePath, TYPE.FILE);
+            }
         }
 
-        diskResourceServiceFacade.getStat(diskResourceUtil.asStringPathTypeMap(list, TYPE.FILE),
+        diskResourceServiceFacade.getStat(pathMap,
```

Comment:
Rather than iterating over the resources to send and building a list of `HasPath` items, which are then converted to a "path:type" map by `diskResourceUtil.asStringPathTypeMap`, I refactored this loop to build the `pathMap` directly. Then the keys (paths) of the `pathMap` are used in the `shareWithAnonymous` call below.

This also automatically de-duplicates paths of genome file index files (*.bai or *.tbi) that need to be shared with anon if the user already selected those files from the Data window (see the `indexFilePath` above).